### ISSUE TITLE
[Lab 2] LSTM "lengths" Parameter

### ISSUE THREAD

**Issue by andretri** [slp-ntua/patrec-labs#55]
Στην μέθοδο ```forward``` (και κατ' επέκταση στη ```last_timestep```) η παράμετρος ```lengths``` τι αντιπροσωπεύει; Το Doc-string της ```forward``` γράφει πως είναι N x 1, αλλά όχι σε τι χρησιμεύει.

**Comment by georgepar** [comment]
είναι τα μήκη των ακολουθιών του batch πριν το padding.

**Comment by andretri** [comment]
η ακολουθία μπορεί να είναι prepadded ή θα υπάρχουν σφάλματα;

**Comment by georgepar** [comment]
την ακολουθία τη βγάζεις padded από τον Dataloader, αλλά πρέπει να δώσεις τα αρχικά μήκη στο lstm για να μην παρσάρεις το padding.

**Comment by ChristosSgouropoulos** [comment]
Καλησπέρα σας. Έφτιαξα μία συνάρτηση η οποία γεμίζει όλες τις παρατηρήσεις με μηδενικά έτσι ώστε να έχουν το ίδιο shape. Συγκεκριμένα κάθε Χ από τον dataloader μου έχει shape tensor:(batch_size , 153 , 6) όπου 153 είναι το μεγαλύτερο length  και 6 είναι ο αριθμός των mfccs. Τα δεδομένα ειναι scaled και όλα καλά. Όταν τα περάσω από το δίκτυο έτσι , το σκορ που μου βγάζει είναι 0.15. Αυτό καταλαβαίνω ότι συμβαίνει επειδή το loss υπολογίζεται για τα μηδενικά που στην πραγματικότητα βάλαμε εμεις , και αρα διαδίδεται πίσω με λανθασμένο τρόπο. 

Μπορείτε να με βοηθήσετε σε αυτό?  Tι shape θα πρέπει να έχει το x που δίνω στη forward πριν μπει στο LSTM layer , και τι shape θα πρέπει να έχει το x πριν μπει και αφού βγεί από το Linear layer?

Αν μπορείτε να μου απαντήσετε αυτά θα με βοηθήσετε αρκετά για να μπορέσω να συνεχίσω.

Σας ευχαριστώ!

**Comment by georgepar** [comment]
ισως θες να τσεκαρεις αυτο

https://towardsdatascience.com/taming-lstms-variable-sized-mini-batches-and-why-pytorch-is-good-for-your-health-61d35642972e
